test/test_output_writers.jl

```diff
@@ -7,6 +7,58 @@ using Dates: Millisecond
 using Oceananigans: write_output!
 using Oceananigans.BoundaryConditions: PBC, FBC, ZFBC, ContinuousBoundaryFunction
 using Oceananigans.TimeSteppers: update_state!
+using Oceananigans.OutputWriters: construct_output
+
+
+function test_output_construction(model)
+
+    u, v, w = model.velocities
+    set!(model, u=(x, y, z)->y)
+
+    indices1 = (:, :, :)
+    indices2 = (1, 2:3, :)
+
+    # First we test fields
+    @test construct_output(u, model.grid, indices1, false) isa Field
+    @test construct_output(u, model.grid, indices2, false) isa Field
+
+    # Now we test windowed fields
+    u_sliced1 = Field(u, indices=indices1)
+    u_sliced2 = Field(u, indices=indices2)
+
+    @test construct_output(u_sliced1, model.grid, indices1, false) isa Field
+    @test construct_output(u_sliced1, model.grid, indices2, false) isa Field
+
+    @test construct_output(u_sliced2, model.grid, indices1, false) isa Field
+    @test construct_output(u_sliced2, model.grid, indices2, false) isa Field
+
+    @test parent(u_sliced2) != parent(construct_output(u_sliced1, model.grid, indices2, false)) # u_sliced2 should have halo regions
+    @test parent(u_sliced2) == parent(construct_output(u_sliced1, model.grid, indices2, true)) # both should have halo regions
```

Comment:
Why do we test `parent` here?

Reply:
It kept throwing me a `true` result even when the indices were different. Not sure why the comparison didn't go deep enough

Reply:
That's certainly possible if the field is full of zeros, and the two indices have the right dimensions. I see we set the field to `y` but this might not matter if the grid has 1 point in y.